src/pages/questions/useQuestionBuffer.ts

```diff
@@ -87,21 +143,58 @@ function reducer(state, action) {
         answers: [
           ...state.answers,
           {
-            insight_id: answeredQuestion?.insight_id,
-            barcode: answeredQuestion?.barcode,
-            insight_type: answeredQuestion?.insight_type,
-            value: answeredQuestion?.value,
+            ...answeredQuestion,
             validationValue: action.payload.value,
+            isPending: true,
+            sendingTime: new Date().getTime() + action.payload.pendingDelay,
           },
         ],
         remainingQuestionNb: state.remainingQuestionNb - 1,
         // skipped ids is used to correctly compute remainingQuestionNb after each data fetching
         skippedIds: [
           ...state.skippedIds,
-          ...(action.payload.value === -1 ? [action.payload.insight_id] : []),
+          ...(action.payload.value === -1 ? [action.payload.insightId] : []),
         ],
       };
 
+    case "sendAnswers":
+      const minDate = new Date().getTime();
+
+      const newAnswers = state.answers.map((answer) => {
+        const { sendingTime, isPending, validationValue, insight_id } = answer;
+        if (isPending && sendingTime <= minDate) {
+          if (validationValue !== -1) {
+            robotoff.annotate(insight_id!, validationValue);
+            return { ...answer, isPending: false };
+          }
+        }
+        return answer;
+      });
+      return { ...state, answers: newAnswers };
+
+    case "revertAnswer":
+      const answeredQuestionIndex = state.answers.findIndex(
+        ({ insight_id }) => insight_id === action.payload.insightId
+      );
+      const answer = state.answers[answeredQuestionIndex];
```

Comment:
not sure if this could happen, so maybe it is not relevant,... could it happen that it doesn't find it and answeredQuestionIndex === -1? (so that answer === undefined), maybe it is more evident if using answers.find( ... ) instead of answers.findIndex() as the former returns an item or undefined

Reply:
It should not occurs, because of the current UI. Only yes/no questions are displayed and so can be reverted

But better to have an early return if no answer correspond to the insight id